impress-org/givewp
# Anonymous donation and company setting choices are now persisted

Resolves #5632 

## Description

This PR introduces logic to ensure that anonymous giving and company name settings for donors are persisted correctly. Previously, the Anonymous GIving option did not correctly persist the donor's choice, and there was no control for a donor to update their company name. This PR introduces logic to address both issues.

## Affects
This PR affects frontend rendering logic for the Donor Profile (Edit Profile tab), and backend logic for updating Donor Profiles (introduction of UpdateDonorAnonymousGiving and UpdateDonorCompany stages in the Edit Profile pipeline).

## Visuals

<img width="681" alt="Screen Shot 2021-02-22 at 5 08 34 PM" src="https://user-images.githubusercontent.com/5186078/108776983-21738c00-7531-11eb-980f-aa243883802f.png">

<img width="676" alt="Screen Shot 2021-02-22 at 5 08 21 PM" src="https://user-images.githubusercontent.com/5186078/108776989-23d5e600-7531-11eb-874d-b87265e65f72.png">

## Pre-review Checklist

## Testing Instructions

From a logged in Donor Profile:
1) Go to the Edit Profile tab
2) Update donor's company name
3) Update donor's anonymous giving preference
4) Refresh, and navigate again to Edit Profile
5) Check that changes were persisted correctly
